src/index.c

```diff
@@ -1169,6 +1169,9 @@ int git_index_remove_bypath(git_index *index, const char *path)
 		ret != GIT_ENOTFOUND))
 		return ret;
 
+	if (ret == GIT_ENOTFOUND)
+		giterr_clear();
+
```

Comment:
Do we need the `if` here? We already return if we find a different error code, so ret here could only be 0 or `ENOTFOUND`, and we should clear the error for both, since we're returning 0 in both of these cases.